src/Capability/Reader/Internal/Class.hs

```diff
@@ -88,17 +92,25 @@ reader = reader_ (proxy# @_ @tag)
 {-# INLINE reader #-}
 
 -- | Execute the given reader action on a sub-component of the current context
--- as defined by the given transformer @t@.
+-- as defined by the given transformer @t@, retaining arbitrary capabilities.
 --
 -- See 'Capability.State.zoom'.
 --
 -- This function is experimental and subject to change.
 -- See <https://github.com/tweag/capability/issues/46>.
-magnify :: forall outertag innertag t outer inner m a.
+magnify :: forall outertag innertag t (cs :: [(* -> *) -> Constraint]) outer inner m a.
```

Comment:
Out of curiosity, how does `magnify` interact with `local` if we're "aliasing" the same component of a reader environment? E.g. in
```
magnify @"foobar" @"foo" @(Rename 1 :.: Pos 1 "foobar") @('[HasReader "foobar" (Int, Int)]) $ do
    ab <- local @"foo" (const 1) (ask @"foobar")
    a <- local @"foobar" (const (2, 3)) (ask @"foo")
    pure (ab, a)
```

Reply:
This is well behaved: the changes passed to `local` are reflected in the outer structure as well, and vice versa. I have added an example to the test suite.